ui/src/datasource/DatasourceForm.vue

```diff
@@ -47,16 +50,36 @@
           @stepChanged="dialogStep = $event"
         />
       </v-stepper-content>
+      <v-stepper-content 
+        v-if="!updateIsSchemaAlive" 
+        step="2"
+      >
+        <adapter-config
+          v-model="datasource"
+          @validityChanged="validStep2 = $event+1"
+        />
+        <stepper-button-group
+          :step="2"
+          :next-enabled="validStep2"
+          @stepChanged="dialogStep = $event+1"
```

Comment:
When I saw it right, the only difference here is that you add `+1` to skip the next step, right? We should be able to remove the duplicate by using an expression like `isSchemaAlive? $event : $event + 1`